glmtools/glm4d.py

```diff
@@ -54,4 +68,24 @@ def t_test_3d(c, X, B, sigma_2, df):
         two-tailed probability value for each t statistic.
     """
     # Your code code here
-    return
+
+    B_2d = B.reshape((-1, B.shape[-1]))
+
+    c_b_cov = c.dot(npl.pinv(X.T.dot(X))).dot(c)
```

Comment:
Aha - I was expecting you to use the `glm` function to run the 2D analysis.   The problem with that, I suspect, is that your 2D analysis doesn't allow more than one column in your Y matrix (see below).